twitcher/oauth2.py

```diff
@@ -0,0 +1,253 @@
+# coding: utf-8
+"""
+twitcher.oauth2
+---------------
+
+Using OAuth2 client_credentials grant type:
+https://docs.apigee.com/api-platform/security/oauth/oauth-20-client-credentials-grant-type
+
+Code examples taken from:
+
+* https://github.com/thomsonreuters/bottle-oauthlib
+* https://github.com/lepture/flask-oauthlib
+
+Example usage:
+
+Get Token::
+
+    http://localhost:8000/oauth/token?grant_type=client_credentials&client_id=alice&client_secret=secret
+
+Use Token::
+
+    http://localhost:8000/ows/proxy/emu?access_token=TOKEN&service=wps&request=Execute&version=1.0.0&DataInputs=name=Stranger
+
+TODO:
+
+* use external login service (github, esgf, ...)
+* use external oauth service for compute
```

Comment:
Seems like Twitcher starts to have functionalities closer and closer to methods already implemented in Magpie.
Any reason not to use it?

Reply:
Need to implement the demands for ESGF and Copernicus. See also Keycloak:
https://github.com/bird-house/twitcher/issues/87

But in a later re-design we can try to avoid duplicate implementations.